src/Files.App/Actions/Navigation/ReopenClosedTabAction.cs

```diff
@@ -0,0 +1,59 @@
+﻿using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.DependencyInjection;
+using Files.App.Commands;
+using Files.App.Contexts;
+using Files.App.Extensions;
+using Files.App.UserControls.MultitaskingControl;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Linq;
+using System.Threading.Tasks;
+using Windows.System;
+
+namespace Files.App.Actions
+{
+	internal class ReopenClosedTabAction : ObservableObject, IAction
+	{
+		private readonly IMultitaskingContext context = Ioc.Default.GetRequiredService<IMultitaskingContext>();
+
+		public string Label => "ReopenClosedTab".GetLocalizedResource();
+
+		public string Description => "TODO: Need to be described";
+
+		public HotKey HotKey { get; } = new(VirtualKey.T, VirtualKeyModifiers.Control | VirtualKeyModifiers.Shift);
+
+		public bool IsExecutable =>
+			context.Control is not null &&
+			!BaseMultitaskingControl.IsRestoringClosedTab &&
+			BaseMultitaskingControl.RecentlyClosedTabs.Any();
+
+		public ReopenClosedTabAction()
+		{
+			context.PropertyChanged += Context_PropertyChanged;
+			BaseMultitaskingControl.IsRestoringTabChanged += BaseMultitaskingControl_IsRestoringTabChanged;
+			BaseMultitaskingControl.RecentlyClosedTabs.CollectionChanged += RecentlyClosedTabs_CollectionChanged;
```

Comment:
Can these subscribe to the same event?

Reply:
You mean the two `BaseMultitaskingControl`?
We could but we should invoke the event manually each time we add or remove items from `RecentlyClosedTabs` (which then should be converted to a `List`)

Reply:
I mean `BaseMultitaskingControl_IsRestoringTabChanged` and `RecentlyClosedTabs_CollectionChanged.